src/app.ts

```
@@ -85,9 +85,7 @@ const processPullRequest = async (payload: PullRequestEvent, context: Context<'p
 
     log.info(`Validated configuration changes in #${number} with conclusion: ${checkConclusion}.`)
   } catch (error) {
-    log.error(`Failed to process PR #${number}' with error:`)
-    log.error(error as never)
-    throw error
+    log.error(error, `Failed to process PR #${number}`)
```

This is going to make tests in `app.test.ts` pass even if they throw an exception

---

Nope, the test explicitly validates that an error has been logged [here](https://github.com/pleo-oss/file-distributor/pull/54/files#diff-ba6d46dd662f63ed71795ba931c5bfd4260591e76a23126a5956f8e4d0a7d24aR167)

---

That's why we have the overriden logger, so we can validate the expected error 😄

---

In this test yes, but all tests become more brittle. On PR https://github.com/pleo-oss/file-distributor/pull/44 there were many tests passing in spite of `nok` not matching routes, just because an exception would be thrown and swallowed in the `try` `catch`

---

Tried to see whether we can spy on the internal `log` of the `Context`, but my `jest` skills aren't good enouhg 🤔 I'll see if I can make the output nice while throwing, then

---

Done, WDYT?

---

Given that we already log the PR or commit we're processing, it doesn't really make sense to log and re-throw at all. It is after all a bad design pattern 🥵 

I've remove the global try/catch in the event handlers now - the tests are very strict as a result and we'll still get nice error logs - LMK WYT 🙏🏼